docs/en/sql-reference/functions/string-search-functions.md

```diff
@@ -590,8 +592,55 @@ Result:
 └───────────────────────────────┘
 ```
 
-[Original article](https://clickhouse.tech/docs/en/query_language/functions/string_search_functions/) <!--hide-->
-
 ## countMatches(haystack, pattern) {#countmatcheshaystack-pattern}
 
 Returns the number of regular expression matches for a `pattern` in a `haystack`.
+
+**Syntax**
+
+``` sql
+countMatches(haystack, pattern)
+```
+
+**Parameters**
+
+-   `haystack` — The string to search in. [String](../../sql-reference/syntax.md#syntax-string-literal).
+-   `pattern` — The regular expression with [re2 syntax](https://github.com/google/re2/wiki/Syntax). [String](../../sql-reference/data-types/string.md).
+
+**Returned value**
+
+-   The number of matches.
+
+Type: [UInt64](../../sql-reference/data-types/int-uint.md).
+
+**Examples**
+
+Query:
+
+``` sql
+SELECT countMatches('foobar.com', 'o')
```

Comment:
I suggest to replace `'o'` in this example with `'o+'` to provide an example of the pattern being a regular expression.